File: 0019-remove-nth-node-from-end-of-list/0019-remove-nth-node-from-end-of-list.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* removeNthFromEnd(ListNode* head, int n) {
        ListNode* fast = head;
        ListNode* slow = nullptr;
        int cnt = 0;
        while(fast->next != nullptr) {
            fast = fast->next;
            cnt++;
            if(cnt == n) slow = head;
            if(cnt > n) slow = slow->next;
        }
        if(slow == nullptr) return head->next;
        slow->next = slow->next->next;
        return head;
    }
};

File: 0019-remove-nth-node-from-end-of-list/0019-remove-nth-node-from-end-of-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode removeNthFromEnd(ListNode head, int n) {
        ListNode fast = head;
        ListNode slow = null;
        int cnt = 0;
        while(fast.next != null) {
            fast = fast.next;
            cnt++;
            if(cnt == n) slow = head;
            if(cnt > n) slow = slow.next;
        }
        if(slow == null) return head.next;
        slow.next = slow.next.next;
        return head;
    }
}